armi/bookkeeping/db/database.py

```diff
@@ -1463,6 +1463,60 @@ def _applyComponentNumberDensitiesMigration(comps, unpackedData):
             c.p["nuclides"] = nuclides
             c.p["numberDensities"] = numberDensities
 
+    @staticmethod
+    def getCycleNodeAtTime(dbPath, startTime, endTime, errorIfNotExactlyOne=False):
+        """Given the path to an ARMI database file and a start and end time (in years), return the full set of all time
+        nodes that correspond to that time period in the database.
+
+        Parameters
+        ----------
+        dbPath : str
+            File path to an ARMI database.
+        startTime : int
+            In years, start of the desired interval.
+        endTime : int
+            In years, end of the desired interval.
+        errorIfNotExactlyOne : boolean
+            Raise an error if more than one cycle/node combination is returned. Default is False.
+
+        Returns
+        -------
+        list of strings
+            A list of strings to the desired time interval, e.g.: ["c01n08", "c14n18EOL"]
+        """
+        # basic sanity checks
+        assert startTime >= 0.0, f"The start time cannot be negative: {startTime}."
+        assert endTime >= startTime, f"The end time ({endTime}) is not greater than the start time ({startTime})."
+
+        # open the H5 file directly
+        h5 = h5py.File(dbPath, "r")
+
+        # read time steps in H5 file
+        thisTime = 0.0
+        cycleNodes = []
+        for h5Key in h5.keys():
+            if h5Key == "inputs":
+                continue
+
+            thisTime = h5[h5Key]["Reactor"]["time"][0]
+            if thisTime >= endTime:
+                cycleNodes.append(h5Key)
```

Comment:
i dont think we would want to append in this case since its after the end time

Reply:
This was here because if the target time is between the start and end time in a given time node, we would want that time node counted. It's just how the loop is structured.